Lib/random.py

```diff
@@ -338,7 +334,8 @@ def randint(self, a, b):
 
     def choice(self, seq):
         """Choose a random element from a non-empty sequence."""
-        # raises IndexError if seq is empty
+        if not seq:
```

Comment:
@rhettinger : This change prevents operation on numpy arrays: `if not seq` will produce a `ValueError` (`ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`). While it might be perverse to use `random` from stdlib on a numpy array, such code exists in the wild, such as in the test suite of scikit-learn which this change then breaks. Perhaps the test is better `seq is None or len(seq) == 0`, since that's what is actually needed for the next line about an empty sequence?